src/key/tests.rs

```diff
@@ -742,3 +742,133 @@ fn test_key_builder_core_param_panic() {
             .param(1, Value::Null)
             .build();
 }
+
+#[test]
+fn test_key_canonicalize() {
+    struct TestCase {
+        key_data: &'static str, // hex
+        canonical_key: CoseKey,
+        deterministic_key: CoseKey,
```

Comment:
What does "deterministic" mean here? How does it relate to "canonical"?

Reply:
RFC 7049 talks about "canonical CBOR", RFC 8949 talks about "deterministic encoding".

Reply:
Yeah, but RFC 7049 is obsoleted by RFC 8949, so both are deterministic now. (And it says "The terms are essentially interchangeable".)

We don't use the term deterministic elsewhere in the code.

Reply:
"Deterministic" isn't mentioned elsewhere in the code because CBOR/COSE spec tried quite hard to avoid it being necessary &ndash; they just missed that `COSE_Key` is likely to end up being signed-over in chains.

Moved to use `rfcNNNN_` prefixes.